### ISSUE TITLE
GLTFDocument.generate_scene crashes when called a second time

### ISSUE THREAD

**Issue by Calandiel** [godotengine/godot#69814]
### Godot version

4.0 beta 8

### System information

Fedora Linux 36 (Workstation Edition)

### Issue description

Calling generate_scene on the same document/state pair crashes with following errors:
```
E 0:00:00:0471   _ready: Parameter "root" is null.
  <C++ Source>   modules/gltf/gltf_document.cpp:6900 @ generate_scene()
  <Stack Trace>  node.gd:14 @ _ready()
```
Doing the same C# results in *no* error messages whatsoever and a silent crash instead.
I was generating gltf scenes with the following code:
```
extends Node

func _ready():
	var source = ProjectSettings.globalize_path("res://debug.gltf")
	var doc = GLTFDocument.new()
	var state = GLTFState.new()
	print(source)
	doc.append_from_file(source, state)
	
	var scene_a = doc.generate_scene(state)
	add_child(scene_a)
	var scene_b = doc.generate_scene(state) # <-- the crash happens here
	add_child(scene_b)
```
Admittedly, I could simply be misunderstanding the intended uses of these classes (the documentation is relatively sparse for them) so if generating scenes on the same pair of doc/state objects isn't supported this wouldn't be a bug, though, in that case, I believe the documentation ought to mention it and a better error message  should be displayed.

### Steps to reproduce

Run the attached project (node.tscn) and notice the raised errors.

### Minimal reproduction project

[gltf_crash.zip](https://github.com/godotengine/godot/files/10196654/gltf_crash.zip)

**Comment by stmSi** [comment]
could not reproduce this in current `master`... @Calandiel can u try again in beta 11?

**Comment by Calandiel** [comment]
> could not reproduce this in current `master`... @Calandiel can u try again in beta 11?

I ran it again on the latest release candidate (rc1) and the error still persists (I checked the minimum reproduction project I originally attached and it results in the same error).
@stmSi  Could you perhaps try running it yourself again? If the error wasn't present in beta 11 but was reintroduced since that could hint at what's causing it. Hopefully it's not OS dependent.

**Comment by lyuma** [comment]
In the original design, `generate_scene` was only intended to be called once, so other usecases were never tested.

I don't deny this is a bug, but for 4.0, the only supported behavior will be calling `generate_scene` once per `GLTFDocument` instance. We'll try and improve this some time in the future, since crashes are no fun.

**Comment by EAinsley** [comment]
Could not reproduce it on v4.3.dev3.official [36e943b6b]. 
Was this already fixed by #74018?

**Comment by akien-mga** [comment]
I tested the MRP in a few versions.

In 4.0-stable and 4.1-stable it produces this error:
```
/home/akien/tmp/godot/bug/gltf_crash/debug.gltf
ERROR: Parameter "root" is null.
   at: generate_scene (modules/gltf/gltf_document.cpp:7252)
ERROR: Parameter "p_child" is null.
   at: add_child (scene/main/node.cpp:1391)
```
No crash on my end, but I haven't checked earlier pre-4.0-stable releases which might have had a worse issue.

In 4.2-stable and 4.3-beta2 there's no error anymore, so I'll close this as fixed in 4.2.
